test/src/executioners/FixedPoint.C

```diff
@@ -0,0 +1,95 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FixedPoint.h"
+
+#include "NonlinearSystemBase.h"
+
+template <>
+InputParameters
+validParams<FixedPoint>()
+{
+  InputParameters params = emptyInputParameters();
+
+  params.addParam<unsigned int>("fp_max_its", 50, "Max Linear Iterations");
+  params.addParam<Real>("fp_abs_tol", 1.0e-50, "Nonlinear Absolute Tolerance");
+  params.addParam<Real>("fp_rel_tol", 1.0e-8, "Nonlinear Relative Tolerance");
+  params.addParam<Real>("fp_abs_step_tol", 1.0e-50, "Nonlinear Absolute step Tolerance");
+  params.addParam<Real>("fp_rel_step_tol", 1.0e-50, "Nonlinear Relative step Tolerance");
+  params.addParamNamesToGroup("fp_max_its fp_abs_tol fp_rel_tol fp_abs_step_tol fp_rel_step_tol",
+                              "FixedPoint");
+  return params;
+}
+
+FixedPoint::FixedPoint(Executioner * ex)
+  : SolveObject(ex),
+    _fp_problem(dynamic_cast<FixedPointProblem &>(_problem)),
+    _fp_max_its(getParam<unsigned int>("fp_max_its")),
+    _fp_abs_tol(getParam<Real>("fp_abs_tol")),
+    _fp_rel_tol(getParam<Real>("fp_rel_tol")),
+    _fp_abs_step_tol(getParam<Real>("fp_abs_step_tol")),
+    _fp_rel_step_tol(getParam<Real>("fp_rel_step_tol"))
+{
+}
+
+bool
+FixedPoint::solve()
+{
+  // evaluate kernels on previous solution
+  _fp_problem.computeFullResidual(*_nl.currentSolution(), _nl.RHS());
+  Real initial_residual_norm = _nl.RHS().l2_norm();
+  Real residual_norm = initial_residual_norm;
+  _console << "Fixed point initial residual norm " << residual_norm << std::endl;
+
+  unsigned int it = 0;
+  for (; it < _fp_max_its; ++it)
+  {
+    _console << "Fixed point iteration " << it << std::endl;
+
+    Real residual_norm_previous = residual_norm;
+    if (!_inner_solve->solve())
```

Comment:
Will this involve a Newton solve?  And line 65 has an extra residual evaluation?

I am asking because I am curious about what is the benefit of the source iteration compared with GMRES. It would be even more interesting to have comparisons for the same problem

Reply:
Never mind. It looks useful for the nonlinear eigenvalue system